### PR TITLE
cleanup of ruby code, remove perl dependency

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

**Comment by thutterer** [comment]
Cool! Reads way more like Ruby :)

File: src/modules/Crowbar.rb [hunk 1]
```diff
@@ -107,6 +103,7 @@ def main
       @modified = false
     end
 
+    # FIXME this should be needed when we remove Perl-json handling
```
Comment (jreidinger) [inline src/modules/Crowbar.rb]: probably it should be "it should not be needed"

Comment (jsuchome) [inline src/modules/Crowbar.rb]: right :-)

File: src/include/crowbar/complex.rb [hunk 2]
```diff
@@ -1043,59 +913,48 @@ def ValidateConduitList(key, event)
...
       i = 0
-      while Ops.less_than(i, 7)
-        items = Builtins.add(
-          items,
-          Item(Id(i), Builtins.tostring(i), i == value)
-        )
-        i = Ops.add(i, 1)
+      while i < 7
+        items << Item(Id(i), i.to_s, i == value)
+        i += 1
```
Comment (mvidner) [inline src/include/crowbar/complex.rb]:
```ruby
items = (0..6).map {|i| Item(Id(i), i.to_s, i == value) }
```

Comment (jsuchome) [inline src/include/crowbar/complex.rb]: thanks

**Pull request merged**